runtime/src/accounts_db.rs

```diff
@@ -1791,17 +2017,30 @@ impl AccountsDB {
             if rvs.is_empty() {
                 storage.set_status(AccountStorageStatus::Full);
 
-                // See if an account overflows the default append vec size.
-                let data_len = (with_meta[infos.len()].1.data.len() + 4096) as u64;
-                if data_len > self.file_size {
-                    self.create_and_insert_store(slot, data_len * 2);
+                // See if an account overflows the append vecs in the slot.
+                let data_len = (with_meta[infos.len()].1.data.len() + STORE_META_OVERHEAD) as u64;
+                if !self.has_space_available(slot, data_len) {
+                    let special_store_size = std::cmp::max(data_len * 2, self.file_size);
+                    if self
+                        .try_recycle_and_insert_store(slot, special_store_size, std::u64::MAX)
```

Comment:
nit: Should we wrap this call to `try_recycle_and_insert_store` in a `Measure` and be add it to the self.stats.store_find_store timing metric?